src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingScriptOperation.cs

```
@@ -218,16 +217,52 @@ private SqlScriptPublishModel BuildPublishModel()
                     "Scripting object count {0}, objects: {1}",
                     selectedObjects.Count(),
                     string.Join(", ", selectedObjects)));
-
+            
+            // Get the server name explicitly after connecting to the server.
+            // This is required to catch the actual server name when targeting sql server on docker in linux.
+            //
+            string server = GetServerNameFromLiveInstance(this.Parameters.ConnectionString);
             string database = new SqlConnectionStringBuilder(this.Parameters.ConnectionString).InitialCatalog;
+
+
             foreach (ScriptingObject scriptingObject in selectedObjects)
             {
-                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(database));
+                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(server, database));
             }
 
             return publishModel;
         }
 
+        private string GetServerNameFromLiveInstance(string connectionString)
+        {
+            SqlConnection connection = new SqlConnection(connectionString);
+            SqlCommand cmd = null;
+            string serverName = string.Empty;
+
+            try {
+                connection.Open();
+            
+                cmd = connection.CreateCommand();
+                cmd.CommandText = "SELECT @@servername";
+
+                object retValue = cmd.ExecuteScalar();  
+                serverName = retValue as string;
+            }
+            catch(Exception e)
+            {
+                Logger.Write(LogLevel.Error, string.Format("Error during SQL query for server name. Message: {0}", e.Message));
+            }
```

So we're swallowing the exception. Should we just let it bubble up? Or rethrow?